src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -421,8 +421,7 @@ def get_object_data(
         for o in objects_api_attachments:
             urls_map[o.variable_key].append(o.document_url)
 
-        for key in dynamic_values.keys():
-            variable = state.get_variable(key)
+        for key, variable in state.variables.items():
```

Comment:
Instead of iterating over the components (e.g. using `iterate_data_with_components`) like I initially wanted for this fix, I can just "invert" the iteration process